src/promptflow/promptflow/executor/_script_executor.py

```diff
@@ -0,0 +1,91 @@
+import asyncio
+import inspect
+import uuid
+
+from pathlib import Path
+from typing import Optional, Mapping, Any
+
+
+from promptflow._utils.logger_utils import flow_logger, logger
+from promptflow._core.flow_execution_context import FlowExecutionContext
+from promptflow._core.operation_context import OperationContext
+from promptflow._core.run_tracker import RunTracker
+from promptflow._core.tool_meta_generator import load_python_module_from_file, PythonLoadError
+from promptflow._core.tracer import Tracer, _traced
+from promptflow._constants import LINE_NUMBER_KEY
+from promptflow.contracts.flow import Flow
+from promptflow.contracts.run_mode import RunMode
+from promptflow.executor._result import LineResult
+from promptflow.storage import AbstractRunStorage
+from promptflow.storage._run_storage import DefaultRunStorage
+
+from .flow_executor import FlowExecutor
+
+
+class ScriptExecutor(FlowExecutor):
+    def __init__(
+        self,
+        entry_file: Path,
+        func: Optional[str],
+        working_dir: Optional[Path] = None,
+        *,
+        storage: Optional[AbstractRunStorage] = None,
+    ):
+        logger.debug("Start initializing the executor.")
+        working_dir = Flow._resolve_working_dir(entry_file, working_dir)
+        m = load_python_module_from_file(entry_file)
+        self._func = getattr(m, str(func), None)
+        if not inspect.isfunction(self._func):
+            raise PythonLoadError(
+                message_format="Failed to load python function '{func}' from file '{entry_file}'.",
+                entry_file=entry_file,
+                func=func,
+            )
+        self._is_async = inspect.iscoroutinefunction(self._func)
+        #  Add 
+        if not hasattr(self._func, "__original_function"):
+            self._func = _traced(self._func)
+        self._storage = storage or DefaultRunStorage()
+
+    def exec_line(
+        self,
+        inputs: Mapping[str, Any],
+        index: Optional[int] = None,
+        run_id: Optional[str] = None,
+        **kwargs,
+    ) -> LineResult:
+        operation_context = OperationContext.get_instance()
```

Comment:
looks like we copied some code from flowExecutor, could we share more code? 

Ideally the execute dag part can be extract as a function in FlowExecutor, and for eager flow it's overrided as calling this part: 
![image](https://github.com/microsoft/promptflow/assets/47586720/caae7c53-617b-4b07-bb27-084bbe80debb)

Reply:
There are a lot of stuffs we can improve, this is the first version to unblock other parts.